raster/r.horizon/main.c

```diff
@@ -233,6 +233,7 @@ int main(int argc, char *argv[])
     parm.bufferzone->description =
         _("For horizon rasters, read from the DEM an extra buffer around the "
           "present region");
+    parm.bufferzone->options = "0-";
```

Comment:
Does the testing framework allow to verify that the module errors out when called with a negative value here and the other places that the option is introduced?

Reply:
I wanted to say yes, there is assertModuleFails, but actually the parameter checking in pygrass throws out ValueError, which gunittest can't deal with.

Reply:
Doesn't pure unittest work https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises?

Reply:
Is pygrass doing the parameter checking also, (reimplemented in Python)? If so, it's kinda expected, but I see the need to have the C code checked, but it's probably more something for tests of the C-based parser specifically.

Reply:
Yes. See #3237 for related discussion.

Reply:
I think it's ready.